src/Microsoft.ML.Data/Transforms/DropSlotsTransform.cs

```diff
@@ -190,98 +240,81 @@ private static VersionInfo GetVersionInfo()
-                _exes[i] = new ColInfoEx(slotDropper, suppressed, typeDst, categoricalRanges);
-            }
-            Metadata.Seal();
-        }
-
-        public static DropSlotsTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
-        {
-            Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
-            h.CheckValue(ctx, nameof(ctx));
-            h.CheckValue(input, nameof(input));
-            ctx.CheckAtModel(GetVersionInfo());
-            return h.Apply("Loading Model", ch => new DropSlotsTransform(h, ctx, input));
+            Host.AssertNonEmpty(ColumnPairs);
+            GetSlotsMinMax(columns, out SlotsMin, out SlotsMax);
+            Host.CheckUserArg(AreRangesValid(SlotsMin, SlotsMax), nameof(columns), "The range min and max must be non-negative and min must be less than or equal to max.");
         }
 
-        private DropSlotsTransform(IHost host, ModelLoadContext ctx, IDataView input)
-            : base(host, ctx, input, null)
+        internal SlotsDroppingTransformer(IHostEnvironment env, ModelLoadContext ctx)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(RegistrationName), ctx)
         {
             Host.AssertValue(ctx);
             // *** Binary format ***
             // <base>
             // for each added column
             //   int[]: slotsMin
             //   int[]: slotsMax (no count)
-            Host.AssertNonEmpty(Infos);
-            var size = Infos.Length;
-            _exes = new ColInfoEx[size];
+
+            Host.AssertNonEmpty(ColumnPairs);
+            var size = ColumnPairs.Length;
+            SlotsMin = new int[size][];
+            SlotsMax = new int[size][];
             for (int i = 0; i < size; i++)
             {
-                int[] slotsMin = ctx.Reader.ReadIntArray();
-                Host.CheckDecode(Utils.Size(slotsMin) > 0);
-                int[] slotsMax = ctx.Reader.ReadIntArray(slotsMin.Length);
-                bool suppressed;
-                ColumnType typeDst;
-                SlotDropper slotDropper = new SlotDropper(Infos[i].TypeSrc.ValueCount, slotsMin, slotsMax);
-                int[] categoricalRanges;
-                ComputeType(input.Schema, slotsMin, slotsMax, i, slotDropper, out suppressed, out typeDst, out categoricalRanges);
-                _exes[i] = new ColInfoEx(slotDropper, suppressed, typeDst, categoricalRanges);
-                Host.CheckDecode(AreRangesValid(i));
-            }
-            Metadata.Seal();
+                SlotsMin[i] = ctx.Reader.ReadIntArray();
+                Host.CheckDecode(Utils.Size(SlotsMin[i]) > 0);
+                SlotsMax[i] = ctx.Reader.ReadIntArray(SlotsMin[i].Length);
+            }
+            Host.Assert(AreRangesValid(SlotsMin, SlotsMax));
+        }
+
+        // Factory method for SignatureLoadModel
```

Comment:
don't forget dot in the end of sentence :)

Reply:
Thank you!

---
In reply to: [236089275](https://github.com/dotnet/machinelearning/pull/1683#discussion_r236089275) [](ancestors = 236089275)